core/src/main/java/org/apache/calcite/rex/RexBuilder.java

```diff
@@ -1871,10 +1872,28 @@ && areAssignable(arg, Arrays.asList(lower, upper))) {
     switch (point.getKind()) {
     case LITERAL:
       final RexLiteral literal = (RexLiteral) point;
+      if (literal.getValue() instanceof AbstractFlatList) {
```

Comment:
Hi @NobiGo I'm not sure if this modification is appropriate. Can you help me to have a look?

Reply:
Can't we use a list here?
 If it is an AbstractFlatList, does it restrict users from using an ImmutableIntList to build it?
 Can only include the integer value?